server/configHelpers.js

```diff
@@ -0,0 +1,185 @@
+// Create a bcfg-compatible config file
+
+const assert = require('bsert');
+const fs = require('bfile');
+const { resolve } = require('path');
+const Config = require('bcfg');
+
+const logger = require('./logger');
+const clientFactory = require('./clientFactory');
+const { loadConfig, getConfigFromOptions } = require('./loadConfigs');
+
+/*
+ * create client config
+ * Note: This will actually create the file in your bpanel prefix location
+ * @param {string} id - id for the client
+ * @param {Object} options object for a bcoin/hsd client
+ * @param {bool} force - whether or not to force config creation if client
+ * can't connect
+ * @returns {bcfg.Config}
+ */
+async function createClientConfig(id, options = {}, force = false) {
+  assert(typeof id === 'string', 'Must pass an id as first paramater');
+  // assert(options, 'Must pass config options to create a client');
+
+  let clientConfig = options;
+  if (!(options instanceof Config))
+    clientConfig = getConfigFromOptions({ id, ...options });
+
+  const appConfig = loadConfig('bpanel', options);
+  const clientsDir = appConfig.str('clients-dir', 'clients');
+
+  // get full path to client configs relative to the project
+  // prefix which defaults to `~/.bpanel`
+  const clientsPath = resolve(appConfig.prefix, clientsDir);
+
```

Comment:
I think we can figure out a way to make this `try/catch->if/else` block be more easy to read. The `if force else throw` part I think could be simplified. What do you think about:

```
await testConfigOptions(clientConfig, { force })
```

and having it catch the error inside the function if force is true

Reply:
any reason to put `force` in an object? I think this could work though.

Reply:
Oh actually, nvm. One of the reasons for the `if` in this function is for the logging to note that we're still creating the config. I actually was doing something like this before, but it felt better to have the `testConfigOptions` be a little purer in that it shouldn't really care whether another function wants to `force` an operation regardless of how the test works out. 

Maybe a better way to fix it would be that `testConfigOptions` should just return `true` if it works or the `ClientErrors` object if it fails. Then when implemented, rather than catch, it just checks the return value.

Reply:
I put `force` in an object because it replicates python kwargs, where the name of the argument describes what it does. If the 2nd argument was just a bool, then without a comment it would be impossible to infer what that argument did without leaving a comment

Reply:
gotcha! I'm still not a fan of dealing with the force logic in `testConfigOptions` as it shouldn't be that functions responsibility, it's just telling you whether the config works or not. 

What do you think about passing true or the error instead? The only downside that I was trying to avoid originally was that the result will always be "truthy" (unless an unrelated error is thrown) and so the calling function would have to differentiate between `true` and if it is an Error object (or check specifically if it's a `ClientsError` object)

Reply:
Gotcha, I'll refactor and see what we can make work.

Reply:
Used the golang idiom to update this and removed the `try/catch->if/else throw`. Let me know what you think!

Reply:
Way easier to read, good job 👍